src/cbc_sdk/platform/jobs.py

```diff
@@ -0,0 +1,223 @@
+            self._refresh()
+        else:
+            self._full_init = True
+
+    @classmethod
+    def _query_implementation(cls, cb, **kwargs):
+        """
+        Returns the appropriate query object for Jobs.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            **kwargs (dict): Not used, retained for compatibility.
+
+        Returns:
+            JobQuery: The query object for this object type.
+        """
+        return JobQuery(cls, cb)
+
+    def _refresh(self):
+        """Reload this object from the server."""
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+        resp = self._cb.get_object(url)
+        self._info = resp
+        self._full_init = True
+        self._last_refresh_time = time.time()
+        return True
+
+    def get_progress(self):
+        """
+        Get and return the current progress information for the job.
+
+        Required Permissions:
+            jobs.status(READ)
+
+        Returns:
+            int: Total number of items to be operated on by this job.
+            int: Total number of items for which operation has been completed.
+            str: Current status message for the job.
+        """
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id) + "/progress"
+        resp = self._cb.get_object(url)
+        self._info['progress'] = resp
+        return resp['num_total'], resp['num_completed'], resp.get('message', None)
+
+    def get_output_as_stream(self, output):
+        """
+        Export the results from the job, writing the results to the given stream.
+
+        Required Permissions:
+            jobs.status(READ)
+
+        Args:
+            output (RawIOBase): Stream to write the CSV data from the request to.
+        """
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id) + '/download'
+        self._cb.get_stream_data(url, output)
+
+    def get_output_as_string(self):
+        """
+        Export the results from the job, returning the results as a string.
+
+        Required Permissions:
+            jobs.status(READ)
+
+        Returns:
+            str: The results from the job.
+        """
+        with io.BytesIO() as buffer:
+            self.get_output_as_stream(buffer)
+            return str(buffer.getvalue(), 'utf-8')
+
+    def get_output_as_file(self, filename):
+        """
+        Export the results from the job, writing the results to the given file.
+
+        Required Permissions:
+            jobs.status(READ)
+
+        Args:
+            filename (str): Name of the file to write the results to.
+        """
+        with io.open(filename, 'wb') as file:
+            self.get_output_as_stream(file)
+
+    def get_output_as_lines(self):
+        """
+        Export the results from the job, returning the data as iterated lines of text.
```

Comment:
Can we add a note that this is intended for csv or text files and should not be used for json files. I would expect that we see more and more file types being used for async jobs

Reply:
Adding that as a documentation change.